backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -428,6 +433,120 @@ public FutureExperimentRunDAO(
         .thenCompose(unused -> deleteExperimentRuns(runIds), executor);
   }
 
+  /**
+   * @param datasets : datasets for privilege check
+   * @param errorOut : Throw error while creation (true) otherwise we will keep it silent (false) at
+   *     fetch time
+   * @return {@link List} : accessible datasets
+   */
+  private InternalFuture<List<Artifact>> filterAndGetPrivilegedDatasetsOnly(
```

Comment:
Move to a handler. This file is already way too big to be maintainable.

Reply:
Ok, Created separate handler for this called `FilterPrivilegedDatasetsHandler` and moved code to it.